### PR TITLE
Avoid including platform BOM twice in the Gradle plugin

### LINKED ISSUE #11251: Gradle addExtension will add BOM dependency even if platform is already in dependencies list
**Describe the bug**
Gradle plugin from master branch will add platform BOM dependency even if `enforcedPlatform` is already in dependency list.

**Expected behavior**
Platform BOM should not be added by addExtension task, if BOM is already present.

**Actual behavior**
From build.gradle, after calling `.\gradlew.bat addExtension --extensions="quarkus-jackson"`
```
dependencies {
    implementation 'io.quarkus:quarkus-jackson'
    implementation 'io.quarkus:quarkus-bom:999-SNAPSHOT'
    implementation enforcedPlatform("${quarkusPlatformGroupId}:${quarkusPlatformArtifactId}:${quarkusPlatformVersion}")
    implementation 'io.quarkus:quarkus-resteasy-jsonb'
    implementation 'io.quarkus:quarkus-resteasy'
    testImplementation 'io.quarkus:quarkus-junit5'
    testImplementation 'io.rest-assured:rest-assured'
}
```

**To Reproduce**
1. Create quarkus gradle project and call  `.\gradlew.bat addExtension --extensions="quarkus-jackson"`

### PR DESCRIPTION
**Pull request by gastaldi**
Fixes #11251

Add test

### CODE REVIEW

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java [hunk 1]
```diff
@@ -61,11 +61,30 @@ protected boolean addDependency(AppArtifactCoords coords, boolean managed) {
...
-                .append(coords.getArtifactId());
+        boolean isBOM = "pom".equals(coords.getType());
+        StringBuilder newDependency;
+        if (isBOM) {
+            // Check if BOM is not included already
+            String resolvedPlatform = String
+                    .format("%s:%s:%s", getProperty(model, "quarkusPlatformGroupId"),
+                            getProperty(model, "quarkusPlatformArtifactId"),
+                            getProperty(model, "quarkusPlatformVersion"));
+            String thisBOM = String.format("%s:%s:%s", coords.getGroupId(), coords.getArtifactId(), coords.getVersion());
```
Comment (gsmet) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java:72]: Do we really want to compare with the version?

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java [hunk 2]
```diff
@@ -61,11 +61,30 @@ protected boolean addDependency(AppArtifactCoords coords, boolean managed) {
     }
 
     static boolean addDependencyInModel(Model model, AppArtifactCoords coords, boolean managed) {
-        StringBuilder newDependency = new StringBuilder()
-                .append("    implementation '")
-                .append(coords.getGroupId())
-                .append(":")
-                .append(coords.getArtifactId());
+        boolean isBOM = "pom".equals(coords.getType());
+        StringBuilder newDependency;
```
Comment (ia3andy) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java:65]: It seems really weird to have the quarkus bom insertion as part of the `addDependencyInModel `

Comment (gastaldi) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java:65]: Why? You can have other BOMs when using the extension registry

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java (hunk 1 above)
Comment (gastaldi) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java:72]: The version will (or should) always be the same here

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java (hunk 2 above)
Comment (ia3andy) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java:65]: aww right, I forgot about that :)

Comment (gastaldi) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java:65]: The extension registry creates an `ExtensionInstallPlan` with the extensions and BOMs needed to work, which then it's up to the `ExtensionManager` to decide what needs to be added/changed/etc

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java (hunk 1 above)
Comment (ia3andy) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java:72]: @gastaldi as @gsmet pointed out, I think we should have the same bahavior as for extensions which is to skip it if it's already there whatever the version.

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java (hunk 2 above)
Comment (ia3andy) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java:65]: @gastaldi do you use the currently installed extensions to create that plan?

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java (hunk 1 above)
Comment (gastaldi) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java:72]: @ia3andy although working with the model as a big String in the case of Gradle is a PITA, I think that might make sense, let's do that in a separate PR

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java (hunk 2 above)
Comment (gastaldi) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java:65]: No, it's up to the `ExtensionManager` to decide what to do with the `ExtensionInstallPlan` and the currently installed extensions

Comment (ia3andy) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java:65]: ok ;-) I just wanted to make sure that my other comment about the version was relevant..

**Review by ia3andy — APPROVED** [review]
LGTM

**Comment by gsmet** [comment]
I would like to have @aloubyansky 's blessing on this one so it will be merged for 1.7.1.Final.

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java [hunk 3]
```diff
@@ -61,11 +61,29 @@ protected boolean addDependency(AppArtifactCoords coords, boolean managed) {
...
+        if (isBOM) {
+            // Check if BOM is not included already
+            String resolvedPlatform = String
+                    .format("%s:%s", getProperty(model, "quarkusPlatformGroupId"),
+                            getProperty(model, "quarkusPlatformArtifactId"));
```
Comment (aloubyansky) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java:70]: It's better but works on the assumption that these properties are used. In general there could be any property or direct coordinates.

**Comment by aloubyansky** [comment]
IMO, it's ok as a quick fix.

**Pull request merged**